src/python/zquantum/core/circuit/gate/_gate.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+import json
+from typing import Tuple, Union, Dict, TextIO
+from collections import Counter
+from ...utils import SCHEMA_VERSION, convert_array_to_dict, convert_dict_to_array
+
+
+class Gate(object):
+    """Class for storing information associated with a quantum gate.
+ 
+    Attributes:
+        matrix (np.ndarray): two-dimensional array defining the matrix representing the quantum operator
+        qubits (tuple[int]): A list of qubit indices that the operator acts on 
+    """
+
+    def __init__(self, matrix: np.ndarray, qubits: Tuple[int]):
+        assert self._is_valid_operator(matrix, qubits)
+        assert self._qubits_are_valid(qubits)
+        self.matrix = matrix
+        self.qubits = qubits
+
+    def _is_valid_operator(self, matrix: np.ndarray, qubits: Tuple[int]):
+        # Make sure matrix is square
+        is_square = True
+        num_rows = len(matrix)
+        for row in matrix:
+            if len(row) != num_rows:
+                is_square = False
+
+        # Make sure matrix is associated with correct number of qubits
+        correct_num_qubits = 2 ** len(qubits) == num_rows
+
+        return is_square and correct_num_qubits
+
+    def _qubits_are_valid(self, qubits: Tuple[int]):
```

Comment:
If `qubits` is just a tuple of integers, why not check if with `set`?
`if len(set(qubits)) != len(qubits)` ?

Reply:
Good idea - was trying to figure out the most efficient way to do this comparison. Added in a future commit